File: apps/web-admin/src/api/device.ts
import request from './request';

import { Playlist } from './playlist';

export interface Device {
  id: string;
  serialNumber: string;
  name: string;
  status: 'online' | 'offline' | 'pending';
  lastHeartbeat: string;
  ipAddress?: string;
  version?: string;
  createdAt: string;
  playlists?: Playlist[];
}

export const getDevices = () => {
  return request.get<any, Device[]>('/devices');
};

File: apps/android-player/src/main/java/com/xplay/player/data/model/Device.kt
package com.xplay.player.data.model

data class Device(
    val id: String,
    val serialNumber: String,
    val name: String,
    val status: String,
    val lastHeartbeat: String?,
    val playlists: List<Playlist> = emptyList()
)

data class RegisterRequest(
    val serialNumber: String,
    val name: String? = null,
    val version: String? = null,
    val ipAddress: String? = null
)
